lib/matplotlib/lines.py

```diff
@@ -790,6 +790,9 @@ def draw(self, renderer):
                     cap = self._dashcapstyle
                     join = self._dashjoinstyle
                 else:
+                    print(tpath.vertices.size)
+                    tpath = affine.inverted().transform_path(affine.transform_path(tpath).cleaned(simplify=True))
+                    print(tpath.vertices.size)
```

Comment:
Copying a comment that I haven't pushed yet:

```python
# the path simplification code expects the path
# to be in pixel space, not data space, but the
# renderer downstream expects the path to be
# in data space. The transformation `affine`
# will put the path into pixel space, and
# affine.inverted() will transform it back to
# data space. So, transform using affine, simplify
# the path, and transform using the affine inverted.
```

It is inefficient, but I see no way around it at the moment. Do you know what populates the `renderer` argument? The only way to delay cleaning until it has been transformed is to do it inside this call: `renderer.draw_path(gc, path, trans)` from

```python
    def _draw_solid(self, renderer, gc, path, trans):
        gc.set_linestyle('solid')
        gc.set_dashes(self._dashOffset, self._dashSeq)
        renderer.draw_path(gc, path, trans)
```